src/constructs/aws/StaticWebsite.ts

```diff
@@ -1,333 +1,51 @@
-import { Bucket } from "@aws-cdk/aws-s3";
-import {
-    AllowedMethods,
-    CachePolicy,
-    Distribution,
-    FunctionEventType,
-    HttpVersion,
-    ViewerProtocolPolicy,
-} from "@aws-cdk/aws-cloudfront";
 import * as cloudfront from "@aws-cdk/aws-cloudfront";
+import { FunctionEventType } from "@aws-cdk/aws-cloudfront";
 import type { Construct as CdkConstruct } from "@aws-cdk/core";
-import { CfnOutput, RemovalPolicy } from "@aws-cdk/core";
-import type { FromSchema } from "json-schema-to-ts";
-import chalk from "chalk";
-import { S3Origin } from "@aws-cdk/aws-cloudfront-origins";
-import * as acm from "@aws-cdk/aws-certificatemanager";
-import { flatten } from "lodash";
 import type { AwsProvider } from "@lift/providers";
-import { AwsConstruct } from "@lift/constructs/abstracts";
-import type { ConstructCommands } from "@lift/constructs";
-import type { Progress } from "../../utils/logger";
-import { getUtils } from "../../utils/logger";
-import { s3Sync } from "../../utils/s3-sync";
-import ServerlessError from "../../utils/error";
-import { emptyBucket, invalidateCloudFrontCache } from "../../classes/aws";
+import type { FromSchema } from "json-schema-to-ts";
 import { redirectToMainDomain } from "../../classes/cloudfrontFunctions";
+import { getCfnFunctionAssociations } from "../../utils/getDefaultCfnFunctionAssociations";
+import { COMMON_STATIC_WEBSITE_DEFINITION, StaticWebsiteAbstract } from "./abstracts/StaticWebsiteAbstract";
 
 const STATIC_WEBSITE_DEFINITION = {
     type: "object",
     properties: {
-        type: { const: "static-website" },
-        path: { type: "string" },
-        domain: {
-            anyOf: [
-                { type: "string" },
-                {
-                    type: "array",
-                    items: { type: "string" },
-                },
-            ],
-        },
-        certificate: { type: "string" },
-        security: {
-            type: "object",
-            properties: {
-                allowIframe: { type: "boolean" },
-            },
-            additionalProperties: false,
-        },
-        errorPage: { type: "string" },
+        ...COMMON_STATIC_WEBSITE_DEFINITION.properties,
         redirectToMainDomain: { type: "boolean" },
```

Comment:
Any reason why this property isn't on the new construct as well?

Reply:
Since the implementation of the feature enabled `redirectToMainDomain` is already done in a viewer request function, I removed it from the SPA construct to simplify the code. I also thought that this feature was not needed for a SPA.
I can modify the code to add it if you think it is still useful :wink:

Reply:
Oh I would imagine it's useful for SPAs too? That feature is IMO kind of always needed, because websites can be typed in browsers by humans with `www.` and without (because there's no real standard, humans are used to both formats).

So I would expect almost all websites to redirect to their main domain (either with or without `www`)?

Reply:
Done !